pkg/lib/ldap/factory.go

```diff
@@ -0,0 +1,40 @@
+package ldap
+
+import (
+	"github.com/authgear/authgear-server/pkg/api"
+	"github.com/authgear/authgear-server/pkg/lib/config"
+)
+
+type ClientFactory struct {
+	Config       *config.LDAPConfig
+	SecretConfig *config.LDAPServerUserCredentials
+}
+
+func (f *ClientFactory) Authenticate(serverName string, username string, password string) (*Entry, error) {
+	var serverConfig *config.LDAPServerConfig
+	for _, s := range f.Config.Servers {
+		if s.Name == serverName {
+			serverConfig = s
+			break
+		}
+	}
+
+	if serverConfig == nil {
+		return nil, api.ErrLDAPServerNotFound
```

Comment:
I think we will not encounter this error. The JSON schema is supposed to validate the server_name matches one of the defined server. Please update the JSON schema if it is not the case.

So no need to return an API error here. You can just panic.

Reply:
Here i just take reference to existing `OAuthProviderFactory`

Reply:
And I will not make such assumption here because this will assumed the method will only be used in authflow api